Lesson_07/homework_07.py

```diff
@@ -0,0 +1,251 @@
+# task 7
+# Calculate unique symbols in the string.
+def unique_symbols_quantity(text):
+    """Calculate unique symbols in text.
+
+    Args:
+        text (string): Some text
+
+    Returns:
+        int: Quantity of unique symbols
+    """
+    return len(set(text.lower().replace(' ', '')))
+
+
+entered_text = 'The quick brown fox jumps over the lazy dog'
+unique_symbols = unique_symbols_quantity(entered_text)
+
+
+# task 8
+# Calculate sum of all Even number in list 'entered_numbers.
+def even_number(numbers_list):
+    """Calculate sum of all even numbers.
+
+    Args:
+        numbers_list (list): List of numbers.
+
+    Returns:
+        int: Sum of even numbers.
+    """
+    even_numbers_sum = 0
+    for number in numbers_list:
+        if number % 2 == 0:
+            even_numbers_sum += number
+    return even_numbers_sum
+
+
+numbers2 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12]
+even_sum = even_number(numbers2)
+
+
+# task 9
+# Date formatted string.
+# Create function which get date as: Day, Month, Year and by default '.'
+# Function should return string in format 'day.month.year'
+# If user's separator differs from default
+# it should return string '15/03/2023'
+def date_format(*args, sep='.'):
+    """Date formatted string.
+
+    Args:
+        args (tuple): Tuple with three inputs: Day, Month, Year
+        sep (str, optional): _description_. Defaults to ".".
+
+    Returns:
+        string: date in formatted string
+    """
+    date_string = sep.join(map(str, args))
+    return date_string
+
+
+date_string1 = date_format(22, 11, 2003)
+date_string2 = date_format(11, 'April', 2003, sep='/')
+
+# task 10
+# Create function which compare two dictionaries with students courses grades.
+# Function should compare grades for each course
+# and return dictionary with grades difference for each student.
+# Student should n't be displayed, if student isn't in both dictionaries.
+grades_1 = {'John Smith': 90, 'Will Smith': 85, 'Katy Perry': 78,
+            'Jeniffer Aniston': 99}
+grades_2 = {'John Smith': 92, 'Will Smith': 87, 'Katy Perry': 80}
+
+
+def compare_grades(grades1, grades2):
+    """Compare students grades.
+
+    Args:
+        grades1 (int): Student grade from first dictionary.
+        grades2 (_type_): Student grade from second dictionary.
+
+    Returns:
+        dict: Dictionary {'student': grade difference}
+    """
+    key_set = set(grades1.keys()) and set(grades2.keys())
+    diff = {}
+    for key in key_set:
+        diff[key] = grades1[key] - grades2[key]
+    return diff
+
+
+grades_result = compare_grades(grades_1, grades_2)
+
+
+# Create function which checks if number is prime.
+def is_prime_number(number):
+    """Check if number is PRIME.
+
+    Args:
+        number (integer): any number
+
+    Returns:
+        boolean: True or False
+    """
+    for i in range(2, int(number ** 0.5) + 1):
+        if number % i == 0:
+            return False
+        return True
```

Comment:
```suggestion
    for i in range(2, int(number ** 0.5) + 1):
        if number % i == 0:
            return False
        return True
```
print(is_prime_result) -> True ( but 9 is not prime number) - please recheck your algorithm 
Pay attention also to code when return boolean values - usually these conditions could be optimized

Reply:
Updated in commit: f61bf23d85eca83d002c0c3ff63679065222887d
    if number <= 1:
        return False
    if any(number % i == 0 for i in range(2, int(number ** 0.5) + 1)):
        return False
    return True